bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java

```diff
@@ -64,31 +70,31 @@
 public class ShellyCoapHandler implements ShellyCoapListener {
     private final Logger logger = LoggerFactory.getLogger(ShellyCoapHandler.class);
 
-    private final ShellyBaseHandler thingHandler;
-    private final ShellyThingConfiguration config;
-    private final GsonBuilder gsonBuilder;
+    private final ShellyBaseHandler th;
+    private ShellyThingConfiguration config = new ShellyThingConfiguration();
+    private final GsonBuilder gsonBuilder = new GsonBuilder();
     private final Gson gson;
     private String thingName;
+    private boolean discovering = false;
 
-    private @Nullable ShellyCoapServer coapServer;
+    private final ShellyCoapServer coapServer;
     private @Nullable CoapClient statusClient;
-    private @Nullable Request reqDescription;
-    private @Nullable Request reqStatus;
+    private Request reqDescription = new Request(Code.GET, Type.CON);;
+    private Request reqStatus = new Request(Code.GET, Type.CON);
 
     private int lastSerial = -1;
+    private Double lastBrightness = -1.0;
     private String lastPayload = "";
     private Map<String, CoIotDescrBlk> blockMap = new HashMap<>();
     private Map<String, CoIotDescrSen> sensorMap = new HashMap<>();
 
-    public ShellyCoapHandler(ShellyThingConfiguration config, ShellyBaseHandler thingHandler,
-            @Nullable ShellyCoapServer coapServer) {
-        Validate.notNull(coapServer);
-        this.thingHandler = thingHandler;
+    private static final byte[] EMPTY_BYTE = new byte[0];
+
+    public ShellyCoapHandler(ShellyBaseHandler th, ShellyCoapServer coapServer) {
+        this.th = th;
```

Comment:
It was better when this field was named `thingHandler`.

Reply:
that causes a lot of line breaks, because statements getting longer = harder to read

Reply:
A poorly named variable makes code much harder to read than longer statements.

Reply:
changed